marklogic-data-hub/src/main/java/com/marklogic/hub/impl/FlowManagerImpl.java

```diff
@@ -184,18 +223,102 @@ public boolean isFlowExisted(String flowName) {
         return false;
     }
 
+    @Deprecated
+    @Override
     public Map<String, Step> getSteps(Flow flow) {
         return flow.getSteps();
     }
 
+    @Override
+    @Deprecated
     public Step getStep(Flow flow, String stepNum) {
         return flow.getStep(stepNum);
     }
 
+    @Override
+    @Deprecated
     public void setSteps(Flow flow, Map<String, Step> stepMap) {
         flow.setSteps(stepMap);
     }
 
+    /**
+     * For certain step types, when the step is deleted, we need to do some cleanup of artifacts on disk and in the
+     * staging and final databases.
+     *
+     * @param flow
+     * @param removedStep
+     */
+    protected void deleteRelatedStepArtifacts(Flow flow, Step removedStep) {
+        if (removedStep == null) {
+            return;
+        }
+
+        if (removedStep.isMappingStep() && !stepIsReferencedByAFlow(removedStep.getName(), StepDefinition.StepDefinitionType.MAPPING)) {
```

Comment:
Multiple steps can have the same mapping name. 
(Although that's not possible from the UI yet as mapping name is autogenerated to "flow-name-mapping-name")
Better to check for `mappingIsReferencedByAFlow` to avoid UI logic spilling in the core.

Reply:
Each mapping file has different version. Would it be better only clean those files that are not referenced by all the steps across all the flows?

Reply:
I agree. Currently, we do not support a shared Mapping across all the steps/flows.

Reply:
> Not sure what you mean - stepIsReferencedByAFlow is checking the step definition type as well.

It's not possible to share a mapping with multiple flows from UI but that's possible from gradle or core java. So if we have the code to delete mapping in `FlowManager` then we need to check for mappings in all flows. (and use something like `mappingIsReferencedByAFlow` to check that)

The assumption
`final String mappingName = flow.getName() + "-" + removedStep.getName();`
is only applicable in case of QS UI. 
Considering that we should move this logic to the `FlowManagerService` from `FlowManager`

Reply:
I updated this to look at options/mapping/name.